tests/suites/test_suite_ctr_drbg.function

```diff
@@ -50,76 +63,126 @@ exit:
-                             char *add2_string, char *result_str )
+void ctr_drbg_validate_no_reseed( char *add_init_string, char *entropy_string,
+                                  char *add1_string, char *add2_string,
+                                  char *result_string )
 {
-    unsigned char entropy[512];
-    unsigned char add_init[512];
-    unsigned char add1[512];
-    unsigned char add_reseed[512];
-    unsigned char add2[512];
-    mbedtls_ctr_drbg_context ctx;
-    unsigned char buf[512];
-    unsigned char output_str[512];
-    int add_init_len, add1_len, add_reseed_len, add2_len;
-
-    mbedtls_ctr_drbg_init( &ctx );
-    memset( output_str, 0, 512 );
 
-    unhexify( entropy, entropy_string );
-    add_init_len = unhexify( add_init, add_init_string );
-    add1_len = unhexify( add1, add1_string );
-    add_reseed_len = unhexify( add_reseed, add_reseed_string );
-    add2_len = unhexify( add2, add2_string );
-
-    test_offset_idx = 0;
-    TEST_ASSERT( mbedtls_ctr_drbg_seed_entropy_len( &ctx, mbedtls_test_entropy_func, entropy, add_init, add_init_len, 32 ) == 0 );
+    char empty_str[1];
+    empty_str[0] = 0;
+    test_suite_ctr_drbg_validate( RESEED_NEVER, add_init_string,
+                                  strlen( entropy_string ) / 2, entropy_string,
+                                  empty_str, add1_string, add2_string,
+                                  result_string );
+    goto exit;
+}
+/* END_CASE */
 
-    TEST_ASSERT( mbedtls_ctr_drbg_random_with_add( &ctx, buf, 16, add1, add1_len ) == 0 );
-    TEST_ASSERT( mbedtls_ctr_drbg_reseed( &ctx, add_reseed, add_reseed_len ) == 0 );
-    TEST_ASSERT( mbedtls_ctr_drbg_random_with_add( &ctx, buf, 16, add2, add2_len ) == 0 );
-    hexify( output_str, buf, 16 );
-    TEST_ASSERT( strcmp( (char *) output_str, result_str ) == 0 );
+/* BEGIN_CASE */
+void ctr_drbg_validate_pr( char *add_init_string, char *entropy_string,
+                           char *add1_string, char *add2_string,
+                           char *result_string )
+{
+    char empty_str[1];
+    empty_str[0] = 0;
+    test_suite_ctr_drbg_validate( RESEED_ALWAYS, add_init_string,
+                                  strlen( entropy_string ) / 6, entropy_string,
+                                  empty_str, add1_string, add2_string,
+                                  result_string );
+    goto exit;
+}
+/* END_CASE */
 
-exit:
-    mbedtls_ctr_drbg_free( &ctx );
+/* BEGIN_CASE */
+void ctr_drbg_validate_nopr( char *add_init_string, char *entropy_string,
+                             char *add1_string, char *add_reseed_string,
+                             char *add2_string, char *result_string )
+{
+    test_suite_ctr_drbg_validate( RESEED_SECOND, add_init_string,
+                                  strlen( entropy_string ) / 4, entropy_string,
+                                  add_reseed_string, add1_string, add2_string,
+                                  result_string );
+    goto exit;
```

Comment:
Not a blocker, but I'd prefer to change `ctr_drbg_validate` into an auxiliary function that returns `int`, and to have four simple wrapper functions (one for each `RESEED_xxx`) that each call `TEST_ASSERT( ctr_drbg_validate( … ) )`. See https://github.com/ARMmbed/mbedtls/pull/1902#discussion_r207929721